src/python/pants/backend/jvm/tasks/jar_publish.py

```diff
@@ -1042,6 +1042,12 @@ def create_source_jar(self, target, open_jar, version):
     # ends up creating 2 jars one scala and other java both including the java_sources.
 
     def abs_and_relative_sources(target):
+      sources = target.payload.get_field('sources')
```

Comment:
Can you provide some explanation for this? I know you were having trouble with a test.